li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -137,7 +144,23 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener callb
 
   @Override
   public void assign(Collection<TopicPartition> partitions) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    Map<TopicPartition, ClusterDescriptor> topicPartitionToClusterMap =
+        _mdsClient.getClustersForTopicPartitions(_clientId, partitions, _mdsRequestTimeoutMs);
+
+    // Reverse the map so that we can have per-cluster topic partition sets.
+    Map<ClusterDescriptor, Set<TopicPartition>> clusterToTopicPartitionsMap = new HashMap<>();
+    for (Map.Entry<TopicPartition, ClusterDescriptor> entry : topicPartitionToClusterMap.entrySet()) {
+      TopicPartition topicPartition = entry.getKey();
+      ClusterDescriptor cluster = entry.getValue();
+      Set<TopicPartition> newTopicPartitionSet =
+          clusterToTopicPartitionsMap.getOrDefault(cluster, new HashSet<TopicPartition>());
```

Comment:
clusterToTopicPartitionsMap.computeIfAbsent(cluster, {} -> new ...).add(topicPartition)

Reply:
Neat! Was looking for this capability.